src/python/pants/backend/shell/target_types.py

```diff
@@ -393,6 +403,42 @@ class ShellCommandRunTarget(Target):
     )
 
 
+class ShellCommandTestTarget(Target):
+    alias = "experimental_test_shell_command"
```

Comment:
This name is a bit of a mouthful. I'm wondering about naming it `experimental_generic_test`? The "shell" name here (and in `experimental_shell_command`) is a bit misleading, since this can run arbitrary processes, it's not limited to just shell constructs. And it's also easily confused with "tests shell code".

Reply:
I don't have a particular care as to what the name is. Please feel free to just tell me what the name should be.

Reply:
Also if called `generic_test`, should it be in a separate `generic` backend instead of the `shell` backend?

Reply:
We can leave as-is for now, since we have to rename `experimental_shell_command` anyway before we de-experimentalize it.